app/controllers/pieces_controller.rb

```diff
@@ -1,14 +1,12 @@
 class PiecesController < ApplicationController
   before_action :authenticate_user!
-  def show
-    @game = Game.find(params[:game_id])
-    @selected_piece = Piece.find(params[:id])
-  end
-
   def update
     @piece = Piece.find(params[:id])
+    return render json: { update_attempt: 'invalid move' } if @piece.valid_move?(params[:piece][:row_coordinate].to_i, params[:piece][:column_coordinate].to_i) == false
     @piece.update_attributes(piece_params)
-    redirect_to game_path(@piece.game)
+    in_check = @piece.game.in_check?.present? ? true : false
```

Comment:
For this, you don't need the ternary. 

`in_check = @piece.game.in_check?.present?` will give the same result. `present?` returns true or false.